internal/controller/vrg_volsync.go

```diff
@@ -142,13 +142,13 @@ func (v *VRGInstance) reconcilePVCAsVolSyncPrimary(pvc corev1.PersistentVolumeCl
 
 	protectedPVC := v.findProtectedPVC(pvc.Namespace, pvc.Name)
 	if protectedPVC == nil {
-		protectedPVC = newProtectedPVC
-		v.instance.Status.ProtectedPVCs = append(v.instance.Status.ProtectedPVCs, *protectedPVC)
+		protectedPVC = v.addProtectedPVC(newProtectedPVC.Namespace, newProtectedPVC.Name)
 	} else if !reflect.DeepEqual(protectedPVC, newProtectedPVC) {
 		newProtectedPVC.Conditions = protectedPVC.Conditions
-		newProtectedPVC.DeepCopyInto(protectedPVC)
 	}
 
+	newProtectedPVC.DeepCopyInto(protectedPVC)
```

Comment:
So this change looks like a bug fix - we always copy stuff from newProtectedPVC into the instance copy.

But if we always do this, we don't need the reflect.DeepEqual check, since the deep copy should also copy the conditions.

Reply:
Not necessarily. No update is required if the old and new values are the same, and no new conditions have been added.